webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyService.java

```diff
@@ -81,18 +89,38 @@
     @Autowired
     ThirdPartyTMS thirdPartyTMS;
 
-    public PollableFuture asyncSyncMojitoWithThirdPartyTMS(Long repositoryId, String thirdPartyProjectId) {
+    public PollableFuture asyncSyncMojitoWithThirdPartyTMS(Long repositoryId, String thirdPartyProjectId, List<Action> actions, List<String> options) {
         ThirdPartySyncJobInput thirdPartySyncJobInput = new ThirdPartySyncJobInput();
+
         thirdPartySyncJobInput.setRepositoryId(repositoryId);
         thirdPartySyncJobInput.setThirdPartyProjectId(thirdPartyProjectId);
+        thirdPartySyncJobInput.setActions(actions);
+        thirdPartySyncJobInput.setOptions(options);
+
         return quartzPollableTaskScheduler.scheduleJob(ThirdPartySyncJob.class, thirdPartySyncJobInput);
     }
 
-    void syncMojitoWithThirdPartyTMS(Long repositoryId, String thirdPartyProjectId) {
+    void syncMojitoWithThirdPartyTMS(Long repositoryId, String thirdPartyProjectId, List<Action> actions, List<String> options) {
         logger.debug("thirdparty TMS: {}", thirdPartyTMS);
         Repository repository = repositoryRepository.findOne(repositoryId);
-        mapMojitoAndThirdPartyTextUnits(repository, thirdPartyProjectId);
-        uploadScreenshotsAndCreateMappings(repository, thirdPartyProjectId);
+
+        actions.sort((o1, o2) -> Integer.compare(o1.ordinal(), o2.ordinal()));
```

Comment:
order might be clearer with a "if" statement. 

if someone changes the order in enum it won't be obvious that this is impacted and will break silently. Or document the order in the enum + test to enforce order.